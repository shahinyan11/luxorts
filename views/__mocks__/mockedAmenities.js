export default {
  'Amina Heaney': {
    '5e066ad7-4194-4bd4-95ec-0d9715f8159b': {
      id: '5e066ad7-4194-4bd4-95ec-0d9715f8159b',
      description: 'Maiores porro nostrum dolorum sunt.',
      name: 'Nolan Kertzmann',
      status: 'active',
      category: null,
      checked: false,
    },
  },
  'Torrie Sawayn': {
    'c277033f-1683-4d2d-a409-d00b0bbe8201': {
      id: 'c277033f-1683-4d2d-a409-d00b0bbe8201',
      description: 'Nemo magnam dolores quam voluptatibus distinctio unde.',
      name: 'Stephine Rohan',
      status: 'active',
      category: null,
      checked: false,
    },
  },
};
