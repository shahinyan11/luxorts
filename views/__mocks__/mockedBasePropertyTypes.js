export default [
  {
    value: '51e35121-c27e-46fc-9feb-b51dab2490ee',
    label: 'repudiandae',
    description: 'description repudiandae',
  },
  {
    value: '995c5714-185d-4085-8512-690903334f6e',
    label: 'molestias',
    description: 'description molestias',
  },
];
