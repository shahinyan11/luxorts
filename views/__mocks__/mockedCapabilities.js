export default {
  APPROVED: {
    '11da7909-9f4c-45d6-a2a0-8a105b067ed2': {
      id: '11da7909-9f4c-45d6-a2a0-8a105b067ed2',
      description: 'Vel commodi ad incidunt repellendus a molestias delectus nam.',
      name: 'Tonya Wunsch',
      status: 'active',
      checked: false,
    },
    'b84c898e-2176-4ac8-9dfa-54186ccef6cb': {
      id: 'b84c898e-2176-4ac8-9dfa-54186ccef6cb',
      description: 'Maxime nostrum ex iste velit nulla incidunt.',
      name: 'Edwin Greenholt',
      status: 'active',
      checked: false,
    },
  },
};
