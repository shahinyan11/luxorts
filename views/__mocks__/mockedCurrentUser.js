export default {
  id: '5c5d814c-8e06-433d-b006-4119311585bd',
  email: 'evhenii.hryhoriev+10@rubygarage.org',
  publicId: 'c4toxHVi',
  verifiedStatus: 'unverified',
  userType: 'traveller_account',
  userProfile: {
    id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
    about: null,
    address: null,
    avatarUrls: {
      original: '/avatar-img-url',
    },
    dateOfBirth: '1994-01-01',
    drivingLicenseFirstName: 'Evhenii',
    drivingLicenseLastName: 'Test',
    firstName: 'Evhenii',
    gender: null,
    lastName: 'Test',
    phoneNumber: '+299999999',
    street: 'street',
    city: 'city',
    state: 'state',
    zipCode: 'zipCode',
    country: 'country',
  },
  userInvitation: [
    {
      id: '52080e19-c5e0-45bb-af58-edbbe70e78fa',
      created_at: '2022-06-23T06:32:55.588Z',
      email: '12@mail.ru',
      last_sent_at: '2022-06-23T06:32:55.474Z',
      resendable: true,
      status: 'pending',
      user_account_id: '7211ccc2-1fca-4799-b1e1-4005e9fc42be',
    },
  ],
};
