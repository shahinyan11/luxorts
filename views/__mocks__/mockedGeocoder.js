export default [
  {
    streetNumber: '240',
    street: '240 North 7th Avenue',
    city: 'Phoenix',
    state: 'Arizona',
    country: 'United States',
    zipCode: '85007',
    fullAddress: '240 N 7th Ave, Phoenix, AZ 85007, USA',
    longitude: -112.0825682,
    latitude: 33.4507026,
  },
];
