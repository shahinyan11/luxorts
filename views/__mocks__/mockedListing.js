export default {
  id: '0645baf3-945d-4122-8a0b-5ab3e95a9b3e',
  applicationStep: 'accomodation',
  status: 'in_draft',
  userAccount: { id: '6cbf2b68-4a26-41b5-bcd5-161cc37e1420', type: 'hostAccount' },
  listingPropertyType: {
    id: 'ea19f127-bc89-4799-bdbb-346126ac3845',
    typeOfPlace: 'private_place',
    propertyType: {
      id: '51e35121-c27e-46fc-9feb-b51dab2490ee',
      name: 'repudiandae',
      description: 'description repudiandae',
    },
  },
  listingLocation: {
    id: '2b4032ec-a469-4489-b7dc-536d8f9552eb',
    country: 'United States',
    city: 'Phoenix',
    state: 'Arizona',
    street: 'West Maryland Avenue',
    apartmentNumber: '12',
    zipCode: '12345',
    longitude: '-73.97447621784085',
    latitude: '40.764468117826794',
  },
  listingDescription: null,
  listingPricing: {
    pricePerDay: 64,
  },
};
