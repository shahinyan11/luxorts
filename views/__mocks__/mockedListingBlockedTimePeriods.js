export default ['25/07/2022', '26/07/2022', '27/07/2022'];
