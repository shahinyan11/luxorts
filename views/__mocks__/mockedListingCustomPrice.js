export default {
  '01/09/2022': 64,
  '02/09/2022': 64,
  '03/09/2022': 64,
  '04/09/2022': 64,
  '05/09/2022': 64,
};
