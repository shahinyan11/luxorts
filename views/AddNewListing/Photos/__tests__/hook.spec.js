import { renderHook } from '@testing-library/react-hooks';

import checkUserVerification from 'utils/auth/checkUserVerification';

import useParametricSelector from 'hooks/useParametricSelector';

import { fetchListing, fetchListingPhotos } from 'state/concepts/listings/actions';
import { dataDeleteEntity } from 'state/data/actions';
import { store } from '__mocks__/react-redux';
import mockedListing from 'views/__mocks__/mockedListing';
import mockedListingPhotos from 'views/__mocks__/mockedListingPhotos';

import useContainer, { getInitialProps } from '../hook';

jest.mock('utils/auth/checkUserVerification', () => jest.fn(() => true));
jest.mock('hooks/useParametricSelector', () => jest.fn(() => false));
jest.mock('state/concepts/listings/selectors', () => ({
  listingSelector: jest.fn(() => mockedListing),
  listingPhotosSelector: jest.fn(() => mockedListingPhotos),
}));
jest.mock('utils/listing/checkListingCompleteness');

const mockedListingId = 'listingId';
const mockedRouter = {
  query: { listingId: mockedListingId },
};
jest.mock('next/router', () => ({
  useRouter: jest.fn(() => mockedRouter),
}));

const mockedFormik = {
  values: {
    listingPhotos: [],
    listingId: mockedListingId,
    photosForDeleting: [],
  },
  setFieldValue: jest.fn(),
  resetForm: jest.fn(),
};
jest.mock('formik', () => ({
  useFormik: jest.fn(() => mockedFormik),
}));

describe('Photos useContainer hook', () => {
  let result = null;

  beforeEach(() => {
    jest.clearAllMocks();

    ({ result } = renderHook(useContainer));
  });

  it('matches snapshot', () => {
    expect(result.current).toMatchSnapshot();
  });

  describe('onRemovePhotoCallback method', () => {
    it('should saves deleted item when file isn`t present', () => {
      result.current.onRemovePhotoCallback({ id: 'id' });

      expect(mockedFormik.setFieldValue).toHaveBeenCalledWith('photosForDeleting', [
        ...mockedFormik.values.photosForDeleting,
        { id: 'id', delete: true },
      ]);
    });

    it('shouldn`t saves deleted item when file is present', () => {
      mockedFormik.setFieldValue.mockClear();

      result.current.onRemovePhotoCallback({ id: 'id', file: 'file' });

      expect(mockedFormik.setFieldValue).not.toHaveBeenCalled();
    });
  });

  describe('handleUpdateFormikListingPhotos method', () => {
    it('should calls formik.resetForm with listingPhotos when isLoading equals false', () => {
      result.current.handleUpdateFormikListingPhotos();

      const expected = {
        values: {
          ...mockedFormik.values,
          listingPhotos: mockedListingPhotos,
        },
      };

      expect(mockedFormik.resetForm).toHaveBeenCalledWith(expected);
    });

    it('shouldn`t calls formik.resetForm with listingPhotos when isLoading equals true', () => {
      useParametricSelector.mockReturnValue(true);

      ({ result } = renderHook(useContainer));

      mockedFormik.resetForm.mockClear();

      result.current.handleUpdateFormikListingPhotos();

      expect(mockedFormik.resetForm).not.toHaveBeenCalled();
    });
  });

  describe('getInitialProps method', () => {
    const ctx = {
      store,
      query: {
        listingId: mockedListingId,
      },
      req: {},
    };

    it('should dispatches all actions when user is verified and listingId is present', async () => {
      await getInitialProps(ctx);

      expect(store.dispatch).toHaveBeenCalledTimes(3);

      expect(store.dispatch).toHaveBeenNthCalledWith(1, dataDeleteEntity('listingPhoto'));
      expect(store.dispatch).toHaveBeenNthCalledWith(2, fetchListing(mockedListingId));
      expect(store.dispatch).toHaveBeenNthCalledWith(3, fetchListingPhotos(mockedListingId));
    });

    it('should dispatches only general actions when user is verified and listingId isn`t present', async () => {
      await getInitialProps({
        ...ctx,
        query: {},
      });

      expect(store.dispatch).toHaveBeenCalledTimes(1);

      expect(store.dispatch).toHaveBeenNthCalledWith(1, dataDeleteEntity('listingPhoto'));
    });

    it('shouldn`t dispatches any actions when user isn`t verified', async () => {
      checkUserVerification.mockReturnValueOnce(false);

      await getInitialProps(ctx);

      expect(store.dispatch).not.toHaveBeenCalled();
    });

    it('should calls whenComplete callback when is server', async () => {
      await getInitialProps(ctx);

      expect(store.logicMiddleware.whenComplete).toHaveBeenCalled();
    });

    it('shouldn`t calls whenComplete callback when isn`t server', async () => {
      await getInitialProps({
        ...ctx,
        req: null,
      });

      expect(store.logicMiddleware.whenComplete).not.toHaveBeenCalled();
    });
  });
});
