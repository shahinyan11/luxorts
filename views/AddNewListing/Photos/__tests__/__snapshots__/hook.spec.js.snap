// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photos useContainer hook matches snapshot 1`] = `
Object {
  "formik": Object {
    "resetForm": [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "values": Object {
              "listingId": "listingId",
              "listingPhotos": Array [
                Object {
                  "delete": false,
                  "description": "frunvg9pg90xmzcsdk76pdtdh073nm6l0hv109utmtgbegj9ty1hw6cei8wgb8dq8w54h6tdxzykrfklaleojs7h4viaymvbswfab5zn6ex9xdyv8t7vhmlec49pvbxhdowc841s85ij9d4uxej40jl59qaah44s59cs2vol1erwfd0tzba7mswwwxrzkdkoravfyv6i7pfr9xia4gvdxyrwtr77hfrig0bx4uszqpmios0p8an3j1b2yvhlch9",
                  "id": "3bc0e5d1-cee6-4f1b-8cfb-4bc6484334b6",
                  "photoUrls": Object {
                    "medium": "memory://listingphoto/3bc0e5d1-cee6-4f1b-8cfb-4bc6484334b6/photo/medium-578d472f5b2b64fd58b9b058646e2432.jpg",
                    "original": "memory://listingphoto/3bc0e5d1-cee6-4f1b-8cfb-4bc6484334b6/photo/52dde97eb85f168bb7e33724c03935d4.jpg",
                    "small": "memory://listingphoto/3bc0e5d1-cee6-4f1b-8cfb-4bc6484334b6/photo/small-7a2ab0b5d3626b5c6b671d6559eeb93c.jpg",
                  },
                  "position": 1,
                  "src": "memory://listingphoto/3bc0e5d1-cee6-4f1b-8cfb-4bc6484334b6/photo/medium-578d472f5b2b64fd58b9b058646e2432.jpg",
                },
              ],
              "photosForDeleting": Array [],
            },
          },
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    },
    "setFieldValue": [MockFunction],
    "values": Object {
      "listingId": "listingId",
      "listingPhotos": Array [],
      "photosForDeleting": Array [],
    },
  },
  "handleUpdateFormikListingPhotos": [Function],
  "isLoading": false,
  "onBackRoute": "/add-new-listing/services/listingId",
  "onRemovePhotoCallback": [Function],
  "showSkeleton": false,
  "stepStatus": "accomodation",
}
`;
